lib/service-manager.js

```diff
@@ -334,11 +334,15 @@ prototype.getInstanceMetas = function(callback) {
     for (var i in instances) {
       if (!instances.hasOwnProperty(i)) continue;
       var instance = instances[i];
-      debug('%s: %j', iid(instance), instance.containerVersionInfo);
-      metas[instance.id] = instance.containerVersionInfo;
+      var meta = instance.containerVersionInfo;
+      debug('%s: meta %j', iid(instance), meta);
 
-      debug('%s: size %j', iid(instance), instance.cpus);
-      metas[instance.id].size = instance.cpus;
+      if (meta) {
```

Comment:
Why not just change the original assignment to `metas[instance.id] = instance.containerVersionInfo || {};`? That way the drivers don't need to do this sort of check themselves.